src/script/view_model/content/PreferencesDeviceDetailsViewModel.ts

```diff
@@ -0,0 +1,132 @@
+  readonly brandName: string;
+
+  static get SESSION_RESET_STATE() {
+    return {
+      CONFIRMATION: 'confirmation',
+      ONGOING: 'ongoing',
+      RESET: 'reset',
+    };
+  }
+
+  constructor(
+    mainViewModel: MainViewModel,
+    private readonly clientRepository: ClientRepository,
+    private readonly conversationRepository: ConversationRepository,
+    private readonly cryptographyRepository: CryptographyRepository,
+  ) {
+    this.actionsViewModel = mainViewModel.actions;
+    this.selfUser = this.clientRepository.selfUser;
+    this.SessionResetState = PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE;
+    this.logger = getLogger('PreferencesDeviceDetailsViewModel');
+
+    this.activationDate = ko.observable();
+    this.device = ko.observable();
+    this.fingerprint = ko.observableArray([]);
+    this.sessionResetState = ko.observable(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+    this.brandName = Config.getConfig().BRAND_NAME;
+
+    this.device.subscribe(clientEntity => {
+      if (clientEntity) {
+        this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+        this._updateFingerprint();
+        const date = formatTimestamp(clientEntity.time);
+        this.activationDate(t('preferencesDevicesActivatedOn', {date}));
+      }
+    });
+  }
+
+  _updateFingerprint = async (): Promise<void> => {
+    this.fingerprint([]);
+    try {
+      const fingerprint = await this.cryptographyRepository.getRemoteFingerprint(this.selfUser().id, this.device().id);
+      this.fingerprint(fingerprint);
+    } catch (error) {
+      this.logger.warn('Error while trying to update fingerprint', error);
+    }
+  };
+
+  clickOnDetailsClose = (): void => {
+    amplify.publish(WebAppEvents.CONTENT.SWITCH, ContentViewModel.STATE.PREFERENCES_DEVICES);
+    this.device(null);
+  };
+
+  clickOnResetSession = async (): Promise<void> => {
+    this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.ONGOING);
+
+    try {
+      const selfConversationId = this.conversationRepository.self_conversation().id;
+      await this.conversationRepository.reset_session(this.selfUser().id, this.device().id, selfConversationId);
+      window.setTimeout(() => {
+        this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.CONFIRMATION);
+      }, MotionDuration.LONG);
+
+      window.setTimeout(() => {
+        this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+      }, 5000);
+    } catch (error) {
+      this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+      throw error;
+    }
+  };
+
+  clickOnRemoveDevice = async (): Promise<void> => {
+    try {
+      await this.actionsViewModel.deleteClient(this.device());
+      this.clickOnDetailsClose();
+    } catch (error) {
+      this.logger.warn('Error while trying to remove device', error);
```

Comment:
Before, the error was not caught here. Does it have any implications that you are now catching it?